sdk/storage/Azure.Storage.Common/src/StorageRequestValidationPipelinePolicy.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.Core.Http;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Common
+{
+    /// <summary>
+    /// Pipeline policy to verify x-ms-client-request-id and x-ms-client-return-request-id 
+    /// headers that are echoed back from a request match.
+    /// </summary>
+    public class StorageRequestValidationPipelinePolicy : SynchronousHttpPipelinePolicy
```

Comment:
We should make this class `internal` since there's no explicit need for users to ever bother with this